src/main/scala/edu/ie3/util/geo/GeoUtils.scala

```diff
@@ -0,0 +1,341 @@
+*/
+package edu.ie3.util.geo
+
+import edu.ie3.util.exceptions.GeoException
+import org.apache.commons.lang3.ArrayUtils
+import org.locationtech.jts.algorithm.ConvexHull
+import org.locationtech.jts.geom.impl.CoordinateArraySequence
+import org.locationtech.jts.geom.{
+  Coordinate,
+  GeometryCollection,
+  GeometryFactory,
+  LineString,
+  LinearRing,
+  Point,
+  Polygon,
+  PrecisionModel
+}
+import tech.units.indriya.ComparableQuantity
+import tech.units.indriya.quantity.Quantities
+import tech.units.indriya.unit.Units.{METRE, RADIAN}
+
+import java.lang.Math._
+import javax.measure.Quantity
+import javax.measure.quantity.Length
+import scala.collection.immutable.{SortedSet, TreeSet}
+import scala.math.pow
+import scala.util.{Failure, Success, Try}
+
+object GeoUtils {
+  val DEFAULT_GEOMETRY_FACTORY: GeometryFactory =
+    new GeometryFactory(new PrecisionModel(), 4326)
+
+  val EARTH_RADIUS: ComparableQuantity[Length] =
+    Quantities.getQuantity(6378137.0, METRE);
+
+  /** Convert a given [[LineString]] with at least two points into a 'safe to be
+    * compared' [[LineString]] This is done by removing duplicates in the points
+    * in the provided linestring as well as a small change of the start
+    * coordinate if the linestring only consists of two coordinates. For details
+    * on the bug inside [[LineString]] that is addressed here, see
+    * https://github.com/locationtech/jts/issues/531
+    *
+    * @param lineString
+    *   the linestring that should be checked and maybe converted to a 'safe to
+    *   be compared' linestring
+    * @return
+    *   a 'safe to be compared' linestring
+    */
+  def buildSafeLineString(lineString: LineString): LineString =
+    if (lineString.getCoordinates.length == 2)
+      buildSafeLineStringBetweenPoints(
+        lineString.getStartPoint,
+        lineString.getEndPoint
+      )
+    // rebuild line with unique points
+    else {
+      val uniqueCoords: Array[Coordinate] = lineString.getCoordinates.distinct
+      if (uniqueCoords.length == 1)
+        buildSafeLineStringBetweenPoints(
+          lineString.getStartPoint,
+          lineString.getEndPoint
+        )
+      else DEFAULT_GEOMETRY_FACTORY.createLineString(uniqueCoords)
+    }
+
+  /** Build an instance of [[LineString]] between two points that is safe to be
+    * compared even if the provided two points consist of exactly the same
+    * coordinates. This is done by increasing the coordinate of the provided
+    * Point `p1` by a small amount to make it different from Point `p2`. For
+    * details on the bug inside [[LineString]] that is addressed here, see
+    * https://github.com/locationtech/jts/issues/531
+    *
+    * @param p1
+    *   start point of the linestring
+    * @param p2
+    *   end point of the linestring
+    * @return
+    *   a [[LineString]] between the provided points
+    */
+  def buildSafeLineStringBetweenPoints(p1: Point, p2: Point): LineString = {
```

Comment:
Here, points are expected as arguments and then their underlying coordinates are extracted

Reply:
Actually, it does still make sense to use points here, since a point is a geometry consisting of one coordinate

Reply:
You are right! I just copied the old code in that case and didn't notice!